File: sdk/src/expr/cond.ts
import type { Expr } from "../types";
import type { Cond } from "../typed";
import { isScratchValue } from "../typed";
import { expr } from "./builder";

/** `Cond` → wire `Expr` for assert / if_else. */
export function toCond(c: Cond): Expr {
  if (isScratchValue(c)) {
    return expr.ref(c);
  }
  return c;
}

File: go-sdk/typed/cond.go
package typed

import (
	"fmt"

	"github.com/ifx-run/ifx/go-sdk/expr"
)

// ToCondExpr converts a bool ScratchValue or Expr node to wire Expr (assert / if_else).
func ToCondExpr(c interface{}) (expr.Node, error) {
	switch v := c.(type) {
	case ScratchValue:
		return expr.Ref(v.Index), nil
	case expr.Node:
		return v, nil
	default:
		return nil, fmt.Errorf("Cond must be typed.ScratchValue or expr.Node, got %T", c)
	}
}
